website/utils/url_utils.ts

```diff
@@ -43,6 +74,66 @@ const tryDecodeURLList = (url: string, delim = ","): string[] | undefined => {
   return urls;
 };
 
+/**
+ * Parse a string list of comma-separated key:value pairs into
+ * a key-value object.
+ *
+ * @param data The string to parse. Expected to be in the format
+ * "key1:value1,key2:value2,...". Colons and commas in keys or values
+ * should be encoded using `encodeURIComponent`.
+ * @returns An object with the parsed key-value pairs. Key and value strings
+ *  will be decoded using `decodeURIComponent`.
+ */
+export function parseKeyValueList(data: string): Record<string, string> {
+  if (data === "") {
+    return {};
+  }
+  const result: Record<string, string> = {};
+  const keyValuePairs = data.split(",");
+  for (const pair of keyValuePairs) {
+    const splitIndex = pair.indexOf(":");
+    const key = pair.slice(0, splitIndex);
+    const value = pair.slice(splitIndex + 1);
+    result[decodeURIComponent(key).trim()] = decodeURIComponent(value).trim();
+  }
+  return result;
+}
+
+function parseFloat(value: string | undefined): number | undefined {
+  if (value === undefined) {
+    return undefined;
+  }
+  const number = Number.parseFloat(value);
+  return Number.isNaN(number) ? undefined : number;
+}
+
+export function deserializeViewerChannelSetting(
+  channelIndex: number,
+  jsonState: ViewerChannelSettingJson
+): ViewerChannelSetting {
+  const result: ViewerChannelSetting = {
+    match: channelIndex,
+    enabled: jsonState.v === "1",
+    surfaceEnabled: jsonState.i === "1",
+    isovalue: parseFloat(jsonState.iv),
+    surfaceOpacity: parseFloat(jsonState.op),
+    colorizeEnabled: jsonState.cz === "1",
+    colorizeAlpha: parseFloat(jsonState.cza),
+  };
```

Comment:
might be worth a little comment to say defaults for missing fields are expected to be handled downstream of all this?